lib/arrays.md

```diff
@@ -5,44 +5,113 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
+2) use the ".each" method to specify that you what follows the method is to be performed for each array element,
+3) define a variable in verticle bars that will be used as a variable for each iteration of the method, and
+4) type out the operation to be performed for each array element.
+
+For my example, I want each array element to be multiplied by the number 2, and the results displayed:
+array2 = [1, 2, 3, 4, 5, 6, 7, 8]
+array2.each {|z| puts z * 2}
+
 
 # How do you add elements to an Array?
 
-Replace me with your response...
+You add elements to an array by "shoveling" them using the syntax below:
+array2 = [1, 2, 3, 4, 5, 6, 7, 8]
+array2 << 9
+(Result) => [1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+You can also do so by using the ".push" method:
+array2 = [1, 2, 3, 4, 5, 6, 7, 8]
+array2.push 9
+(Result) => [1, 2, 3, 4, 5, 6, 7, 8, 9]
+
 
 # Given the Array `["Laura", "Fiona", "Tori"]`, how would you replace `"Fiona"` with `"Florence"` so that you end up with `["Laura", "Florence", "Tori"]`?
 
-Replace me with your response...
+I would use the following code to replace "Fiona" with "Florence":
+na1 = ["Laura", "Fiona", "Tori"]
+na1.delete_at(1)
+na1.insert(1, "Florence")
+(Result) => ["Laura", "Florence", "Tori"]
 
 # What do the methods `push`, `pop`, `shift`, and `unshift` do?
 
-Replace me with your response...
+1) The ".push" method adds an element of a specified value to the end of the array.
+2) The ".pop" method removes the last element of an array and return its value.
+3) The ".shift" method will remove the remove the first element of an array and return its value.
+4) The ".unshift" method will add a new value to the beginning of an array.
+
+
 
 # How do you determine how many elements are in an Array?
 
-Replace me with your response...
+You can use the "size" or "length" methods to determine the number of elements in an array:
+na2 = [1, 'a', 2, 'b', 3, 'c']
+na2.size
+(Result) => 6
+
 
 # How would you reverse the order of elements in an Array?
 
-Replace me with your response...
+You can reverse the order of an array by using the reverse method:
+na3 = [1, 2, 3, 4, 5, 6, 7]
+na3.reverse
+(Result) => [7, 6, 5, 4, 3, 2, 1]
+
 
 # How would you convert a String `"Sumeet Jain"` into an Array `["Sumeet", "Jain"]`? How about to `["S", "u", "m", "e", "e", "t", " ", "J", "a", "i", "n"]`? How would you convert the Array back into a String?
 
-Replace me with your response...
\ No newline at end of file
+Convert a string "Sumeet Jain" into an array:
+"Sumeet Jain".split(" ")
+(Result) => ["Sumeet", "Jain"]
+
+Convert "Sumeet Jain" into an array where each character is its own element (including the space):
+"Sumeet Jain".split("")
+(Result) => ["S", "u", "m", "e", "e", "t", " ", "J", "a", "i", "n"]
+
+Use the ".join" method to concatenate all array elements:
+na1 = "Sumeet Jain"     => "Sumeet Jain"
+na2 = na1.split("")     => ["S", "u", "m", "e", "e", "t", " ", "J", "a", "i", "n"]
+na2.join                => "Sumeet Jain"             
```

Comment:
The result joins the elements back together without a a space. The reason (I'm guessing) is because it's using the space as a delimiter, and therefore it was excluded from the array as an element.